### ISSUE TITLE
[Feature Request] Sync only N first files of the dir

### ISSUE THREAD

**Issue by SerVB** [syncthing/syncthing-android#1279]
Hello!

Syncing only some first files in the folder is useful on mobile devices because they have relatively small storage.

## Use case:
I have a folder with TV-series on my PC and I want to watch them on my tablet in a train. Because of the road time, I'm able to watch not more than two episodes per day. So I set N=2. When I remove a watched file on the tablet, it removes on the PC. Also, the next episode is downloaded.  
To sum up, I always have two not watched episodes on my tablet.

The same method is used in my podcast app but there is no such free movie app found by me.

**Comment by AudriusButkevicius** [comment]
Sadly no, this is too specific to you to be generally useful.

**Comment by wweich** [comment]
You can set the pull order to oldest first and set the folder specific min free disk space setting so that you have some episodes on your phone, without filling the whole storage space.

**Comment by Catfriend1** [comment]
Good idea. Also, you could write a batch script which ...:
- lists a directory sorted by name, age , etc.
- head -n 2 grabs the first two lines (= files)
- If the target syncthing directory has < 2 files move one of the files from the above determined source directory files into the target.
- delete files on your phone you've watched to let your cyclic running batch script refill with a new one.

**Comment by SerVB** [comment]
@AudriusButkevicius , anyway, thanks.

@wweich , I think it will work only if I have a single "tv-series" folder. What if I have some different ones? Also, I'm not a fan of the idea that the amount of episodes depends on free space.

It would be great at least to set "synced space" for the folder (if one episode is less than 600 MB, 1.2 GB gives two episodes for this series/folder) and use pull order by name (like you suggested). Do you think I should ask for this in the [main repo](https://github.com/syncthing/syncthing/issues)?

@Catfriend1 , thanks. I will use this kind of the workaround for now.

**Comment by SerVB** [comment]
Here is my script: <https://gist.github.com/SerVB/26fd23d9b4e0b8aa57a6169ab1508812>.

**Comment by Catfriend1** [comment]
Wow that's cool, @SerVB may I add a link to that on my wiki?

**Comment by SerVB** [comment]
Yes, please. I will add some description in the script start in two hours :)

**Comment by Catfriend1** [comment]
Link: https://github.com/Catfriend1/syncthing-android/wiki/Sync-N-first-files,-%22get-N-movies-for-watching%22

**Comment by SerVB** [comment]
Updated.
